src/openms/source/ANALYSIS/OPENSWATH/MRMTransitionGroupPicker.cpp

```diff
@@ -103,41 +103,53 @@ namespace OpenMS
   }
 
   void MRMTransitionGroupPicker::calculateBgEstimation_(const MSChromatogram& chromatogram,
-      double best_left, double best_right, double & background, double & avg_noise_level)
+      double best_left, double best_right, double peak_height, double & background, double & avg_noise_level)
   {
     // determine (in the chromatogram) the intensity at the left / right border
-    MSChromatogram::const_iterator it = chromatogram.begin();
-    int nr_points = 0;
-    for (; it != chromatogram.end(); ++it)
+    double intensity_left = 0.0;
+    double rt_apex = 0.0;
+    double intensity_right = 0.0;
+    for (MSChromatogram::const_iterator it = chromatogram.begin() + 1; it != chromatogram.end(); it++)
     {
-      if (it->getMZ() > best_left)
+      MSChromatogram::const_iterator it_prev = it;
+      it_prev--; //previous point
+
+      if (it->getMZ() == best_left)
```

Comment:
I dont think thats a good idea, floating point numbers should not be compared with equal. what if there is a numerical error somewhere?